packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts

```diff
@@ -335,16 +423,97 @@ export const outOfOfficeEntryDelete = async ({ ctx, input }: TBookingRedirectDel
   return {};
 };
 
-export const outOfOfficeEntriesList = async ({ ctx }: { ctx: { user: NonNullable<TrpcSessionUser> } }) => {
-  const outOfOfficeEntries = await prisma.outOfOfficeEntry.findMany({
+type GetOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TOutOfOfficeEntriesListSchema;
+};
+
+export const outOfOfficeEntriesList = async ({ ctx, input }: GetOptions) => {
+  const { cursor, limit, fetchTeamMembersEntries, searchTerm } = input;
+  let fetchOOOEntriesForIds = [ctx.user.id];
+
+  if (fetchTeamMembersEntries) {
+    // Get teams where context user is admin or owner
+    const teams = await prisma.membership.findMany({
+      where: {
+        userId: ctx.user.id,
+        OR: [{ role: MembershipRole.ADMIN }, { role: MembershipRole.OWNER }],
+      },
+      select: {
+        teamId: true,
+      },
+    });
+    if (teams.length === 0) {
+      throw new TRPCError({ code: "NOT_FOUND", message: "user_not_admin_nor_owner" });
+    }
+
+    // Fetch team member user ids
+    const members = await prisma.membership.findMany({
+      where: {
+        teamId: {
+          in: teams.map((team) => team.teamId),
+        },
+        userId: {
+          not: ctx.user.id,
+        },
+      },
+      select: {
+        userId: true,
+      },
+    });
+    if (members.length === 0) {
+      throw new TRPCError({ code: "NOT_FOUND", message: "no_team_members" });
+    }
+    fetchOOOEntriesForIds = members.map((member) => member.userId);
+  }
+
+  const getTotalEntries = await prisma.outOfOfficeEntry.count({
     where: {
-      userId: ctx.user.id,
-      end: {
-        gte: new Date().toISOString(),
+      userId: {
+        in: fetchOOOEntriesForIds,
       },
+      ...(searchTerm && {
+        user: {
+          OR: [
+            {
+              email: {
+                contains: searchTerm,
+              },
+            },
+            {
+              username: {
+                contains: searchTerm,
+              },
+            },
+          ],
+        },
+      }),
     },
-    orderBy: {
-      start: "asc",
+  });
+
+  const outOfOfficeEntries = await prisma.outOfOfficeEntry.findMany({
```

Comment:
Can you not add _count in select for this, instead of having separate queries for count and actual data?

Reply:
The first query gets the total count of matching records, while the second query fetches only limited records (say 10). I tried _count can't be used here.